File: Projeto_2/projeto_2 - Kaella Sales/src/SeguroVeiculo.cpp
#include "../include/SeguroVeiculo.h"
#include <iostream>


SeguroVeiculo::SeguroVeiculo(Cliente& cli, float v, std::string vi) 
   : cliente(cli), valor(v), vigencia(vi) {}

void SeguroVeiculo::CalcularValor(){
    valor = 0;
}

std::string SeguroVeiculo::VerificarVigencia(std::string validez){
    if (validez > vigencia){
    return "Inválido";
    }else{
        return "Válido";
    }
}

File: Projeto_2/src/models/seguro_veiculo.py
from datetime import datetime

class SeguroVeiculo:
    def __init__(self, cliente, valor, vigencia):
        self.cliente = cliente
        self.valor = valor
        self.vigencia = vigencia

    def calcular_valor(self, base_valor, taxa):
        raise NotImplementedError("Este método deve ser sobrescrito pelas subclasses")

    def verificar_vigencia(self):
        data_atual = datetime.now()
        vigencia_data = datetime.strptime(self.vigencia, "%d/%m/%Y")